We use /dev/vc/* since devfs times to keep /dev sorted and clean.

  - Rene Rebe <rene@exactcode.de>

--- xorg-server-1.4.2/hw/kdrive/linux/linux.c.vanilla	2008-08-15 15:43:36.000000000 +0200
+++ xorg-server-1.4.2/hw/kdrive/linux/linux.c	2008-08-15 15:44:03.000000000 +0200
@@ -80,10 +80,10 @@
 	vtno = kdVirtualTerminal;
     else
     {
-	if ((fd = open("/dev/tty0",O_WRONLY,0)) < 0) 
+	if ((fd = open("/dev/vc/0",O_WRONLY,0)) < 0) 
 	{
 	    FatalError(
-		       "LinuxInit: Cannot open /dev/tty0 (%s)\n",
+		       "LinuxInit: Cannot open /dev/vc/0 (%s)\n",
 		       strerror(errno));
 	}
 	if ((ioctl(fd, VT_OPENQRY, &vtno) < 0) ||
@@ -94,7 +94,7 @@
     }
     close(fd);
 
-    sprintf(vtname,"/dev/tty%d",vtno); /* /dev/tty1-64 */
+    sprintf(vtname,"/dev/vc/%d",vtno); /* /dev/vc/1-64 */
 
     if ((LinuxConsoleFd = open(vtname, O_RDWR|O_NDELAY, 0)) < 0)
     {
@@ -111,7 +111,7 @@
      *
      * Why is this needed?
      */
-    LinuxCheckChown ("/dev/tty0");
+    LinuxCheckChown ("/dev/vc/0");
     /*
      * Linux doesn't switch to an active vt after the last close of a vt,
      * so we do this ourselves by remembering which is active now.
@@ -435,7 +435,7 @@
     }
     close(LinuxConsoleFd);                /* make the vt-manager happy */
     LinuxConsoleFd = -1;
-    fd = open ("/dev/tty0", O_RDWR|O_NDELAY, 0);
+    fd = open ("/dev/vc/0", O_RDWR|O_NDELAY, 0);
     if (fd >= 0)
     {
 	memset (&vts, '\0', sizeof (vts));	/* valgrind */
